third_party/terraform/resources/resource_cloudfunctions_function.go

```diff
@@ -91,30 +90,10 @@ func resourceCloudFunctionsFunction() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
-				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
-					value := v.(string)
-
-					if len(value) > 48 {
-						errors = append(errors, fmt.Errorf(
-							"%q cannot be longer than 48 characters", k))
-					}
-					if !regexp.MustCompile("^[a-zA-Z0-9-]+$").MatchString(value) {
-						errors = append(errors, fmt.Errorf(
-							"%q can only contain letters, numbers and hyphens", k))
-					}
-					if !regexp.MustCompile("^[a-zA-Z]").MatchString(value) {
-						errors = append(errors, fmt.Errorf(
-							"%q must start with a letter", k))
-					}
-					if !regexp.MustCompile("[a-zA-Z0-9]$").MatchString(value) {
-						errors = append(errors, fmt.Errorf(
-							"%q must end with a number or a letter", k))
-					}
-					return
-				},
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateRegexp(`^[a-zA-Z][_-a-zA-Z0-9]{0,46}[a-zA-Z0-9]$`),
```

Comment:
Is this sourced from discovery docs, or written by us?

Is `a` a valid name in the API? Looking at our old rules, it would have been allowed but wouldn't be by this rule.

Reply:
lol I was about to ask the same thing. If it's our regex we might want to add a handful of tests for known edge case names that are acceptable as I imagine this will get tweaked over time as most our regexes do.